### PR TITLE
Move of render contant START_HAT into ConstantProvider.

### PR DESCRIPTION
**Pull request by moniika**
## The basics

- [x] I branched from develop
- [x] My pull request is against develop
- [x] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)

### Resolves

Part of #3142

### Proposed Changes

Moves START_HAT constant into rendering ConstantProvider and renames it to ADD_START_HATS to resolve conflict of pre-existing constant START_HAT (that defines start hat path).

### Reason for Changes

START_HAT constant was leftover UI constant from block_render_svg.

### CODE REVIEW

File: core/renderers/common/constants.js [hunk 1]
```diff
@@ -37,6 +37,13 @@ goog.require('Blockly.utils.userAgent');
  */
 Blockly.blockRendering.ConstantProvider = function() {
 
+  /**
+   * Whether to add a 'hat' on top of all blocks with no previous or output
+   * connections. Can be overridden by 'hat' property on Theme.BlockStyle.
+   * @type {boolean}
+   */
+  this.ADD_START_HATS = false;
```
Comment (samelhusseini) [inline core/renderers/common/constants.js]: These constants aren't currently alphabetically sorted. Can we add this close to the START_HAT_HEIGHT and START_HAT_WIDTH.

File: core/renderers/measurables/rows.js [hunk 2]
```diff
@@ -290,8 +290,9 @@ Blockly.utils.object.inherits(Blockly.blockRendering.TopRow,
  * @return {boolean} Whether or not the top row has a left square corner.
  */
 Blockly.blockRendering.TopRow.prototype.hasLeftSquareCorner = function(block) {
-  var hasHat = (block.hat ? block.hat === 'cap' : Blockly.BlockSvg.START_HAT) &&
-    !block.outputConnection && !block.previousConnection;
+  var hasHat = (typeof this.block_.hat !== 'undefined' ?
```
Comment (samelhusseini) [inline core/renderers/measurables/rows.js]: block.hat !== 'undefined' instead of this.block_.hat.

Comment (moniika) [reply core/renderers/measurables/rows.js]: Done.

File: core/renderers/common/constants.js (hunk 1 above)
Comment (moniika) [reply core/renderers/common/constants.js]: Done.

**Pull request merged**